plugins/discovery-ec2/src/main/java/org/opensearch/discovery/ec2/AwsEc2SeedHostsProvider.java

```diff
@@ -228,21 +228,21 @@ protected List<TransportAddress> fetchDynamicNodes() {
     }
 
     private DescribeInstancesRequest buildDescribeInstancesRequest() {
-        final DescribeInstancesRequest describeInstancesRequest = new DescribeInstancesRequest().withFilters(
-            new Filter("instance-state-name").withValues("running", "pending")
+        ArrayList<Filter> filters = new ArrayList<Filter>(
+            List.of(Filter.builder().name("instance-state-name").values("running", "pending").build())
         );
```

Comment:
Style nitpick, but I'd suggest skipping creating the extra `List.of()` since you need a mutable list anyway:


```
ArrayList<> filters = new ArrayList<Filter>();
filters.add(Filter.builder().name("instance-state-name").values("running", "pending").build());
```

not a bit deal either way